tools/online_editor/src/preview_widget.ts

```diff
@@ -122,7 +122,14 @@ export class PreviewWidget extends Widget {
       if (this.#instance == null) {
         this.#instance = component.create(this.canvas_id);
         this.#instance.show();
-        slint.run_event_loop();
+        try {
+          slint.run_event_loop();
```

Comment:
There is, in theory, another way of solving this, by using winit's new spawn function. The main issue with that is the chain of dependencies of routing this through the different crates: wasp-interpreter, slint-interpreter and winit backend.

So we could offer `slint_spawn_event_loop()` in the wasm-interpreter wasm-bindgen API, which would route into `slint_interpreter::spawn_event_loop()` (should be wasm32 cfg guarded), but then where do we go from here? A new method in the backend trait that's wasm32-cfg'ed?

Reply:
Yeah, this breaks quite a lot of abstractions. 
This works for now. We can reconsider a better way if we do an actual web backend.